tasks/market/task_find_deal.go

```diff
@@ -0,0 +1,352 @@
+package market
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	market9 "github.com/filecoin-project/go-state-types/builtin/v9/market"
+	"github.com/filecoin-project/go-state-types/exitcode"
+
+	"github.com/filecoin-project/curio/deps/config"
+	"github.com/filecoin-project/curio/harmony/harmonydb"
+	"github.com/filecoin-project/curio/harmony/harmonytask"
+	"github.com/filecoin-project/curio/harmony/resources"
+	"github.com/filecoin-project/curio/lib/promise"
+
+	"github.com/filecoin-project/lotus/api"
+	apitypes "github.com/filecoin-project/lotus/api/types"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/market"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+const FindDealTaskPollInterval = time.Second * 5
+
+type fdealApi interface {
+	headAPI
+	ChainGetMessage(context.Context, cid.Cid) (*types.Message, error)
+	StateLookupID(context.Context, address.Address, types.TipSetKey) (address.Address, error)
+	StateMarketStorageDeal(context.Context, abi.DealID, types.TipSetKey) (*api.MarketDeal, error)
+	StateNetworkVersion(context.Context, types.TipSetKey) (apitypes.NetworkVersion, error)
+	StateSearchMsg(ctx context.Context, from types.TipSetKey, msg cid.Cid, limit abi.ChainEpoch, allowReplaced bool) (*api.MsgLookup, error)
+}
+
+type FindDealTask struct {
+	db  *harmonydb.DB
+	api fdealApi
+	TF  promise.Promise[harmonytask.AddTaskFunc]
+	cfg *config.MK12Config
+}
+
+func NewFindDealTask(db *harmonydb.DB, api fdealApi, cfg *config.MK12Config) *FindDealTask {
```

Comment:
I just don't understand what this does. Could Market internals have a /documentaion/en/ entry?

Reply:
Once the PSD message is sent to the chain, market deals are created for the proposals. This task finds the deal ID assigned to each proposal we sent in PSD message.